context-propagation/src/main/java/io/micrometer/context/ContextRegistry.java

```diff
@@ -145,6 +145,36 @@ public ContextRegistry registerThreadLocalAccessor(ThreadLocalAccessor<?> access
         return this;
     }
 
+    /**
+     * Removes a {@link ThreadLocalAccessor}.
+     * @param key under which the accessor got registered
+     * @return the same registry instance
+     */
+    public ContextRegistry removeThreadLocalAccessor(String key) {
+        for (ThreadLocalAccessor<?> existing : this.threadLocalAccessors) {
+            if (existing.key().equals(key)) {
+                this.threadLocalAccessors.remove(existing);
+                break;
+            }
+        }
+        return this;
```

Comment:
Consider returning `boolean` instead of `this` to highlight whether the removal was successful or not, rather than acting as a builder method. Same for the `ContextAccessor` counter part.